### PR TITLE
[Stack-hci] Bump up `Stack-hci` API version to `2023-03-01`

### PR DESCRIPTION
**Pull request by Jing-song**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

Migrate `azure-stack-hci` commands to aaz: https://github.com/Azure/aaz/pull/178
Upgrade api-version to 2023-03-01
Add --ids parameter support for Azure Stack HCI
swagger change: https://github.com/Azure/azure-rest-api-specs/pull/23648

close https://github.com/Azure/azure-cli/issues/26020
close https://github.com/Azure/azure-cli/issues/26021
### Related command

`az stack-hci`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Stack-hci

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_create.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting create",
+)
+class Create(AAZCommand):
+    """Create ArcSetting for HCI cluster.
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_create.py:18]:
```suggestion
    """Create arc setting for HCI cluster.
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_create.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+    :example: Create ArcSetting
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_create.py:20]:
```suggestion
    :example: Create arc setting
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_create_identity.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting create-identity",
+)
+class CreateIdentity(AAZCommand):
+    """Create Aad identity for arc settings.
+
+    :example: Create Arc Identity
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_create_identity.py:20]:
```suggestion
    :example: Create arc identity
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_delete.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting delete",
+    confirmation="Are you sure you want to perform this operation?",
+)
+class Delete(AAZCommand):
+    """Delete ArcSetting resource details of HCI Cluster.
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_delete.py:19]:
```suggestion
    """Delete arc setting resource details of HCI Cluster.
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_delete.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+    :example: Delete ArcSetting
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_delete.py:21]:
```suggestion
    :example: Delete arc setting
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_generate_password.py [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting generate-password",
+)
+class GeneratePassword(AAZCommand):
+    """Generate password for arc settings.
+
+    :example: Generate Password
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_generate_password.py:20]:
```suggestion
    :example: Generate password
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_list.py [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting list",
+)
+class List(AAZCommand):
+    """List ArcSetting resources of HCI Cluster.
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_list.py:18]:
```suggestion
    """List arc setting resources of HCI Cluster.
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_list.py [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+
+    :example: List ArcSetting resources by HCI Cluster
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_list.py:20]:
```suggestion
    :example: List arc setting resources by HCI Cluster
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_show.py [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting show",
+)
+class Show(AAZCommand):
+    """Get ArcSetting resource details of HCI Cluster.
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_show.py:18]:
```suggestion
    """Get arc setting resource details of HCI Cluster.
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_update.py [hunk 10]
```diff
@@ -0,0 +1,459 @@
...
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci arc-setting update",
+)
+class Update(AAZCommand):
+    """Update ArcSetting for HCI cluster.
+
+    :example: Update ArcSetting
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/arc_setting/_update.py:20]:
```suggestion
    """Update arc setting for HCI cluster.

    :example: Update arc setting
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_create.py [hunk 11]
```diff
@@ -0,0 +1,410 @@
...
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci extension create",
+)
+class Create(AAZCommand):
+    """Create Extension for HCI cluster.
+
+    :example: Create Arc Extension
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_create.py:20]:
```suggestion
    """Create extension for HCI cluster.

    :example: Create arc extension
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_delete.py [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+
+@register_command(
+    "stack-hci extension delete",
+    confirmation="Are you sure you want to perform this operation?",
+)
+class Delete(AAZCommand):
+    """Delete particular Arc Extension of HCI Cluster.
+
+    :example: Delete Arc Extension
+        az stack-hci extension delete --arc-setting-name "default" --cluster-name "myCluster" --name "MicrosoftMonitoringAgent" --resource-group "test-rg"
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_delete.py:22]:
```suggestion
    """Delete particular arc extension of HCI Cluster.

    :example: Delete arc extension
        az stack-hci extension delete --arc-setting-name "default" --cluster-name "myCluster" --name "MicrosoftMonitoringAgent" --resource-group "test-rg"
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_list.py [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci extension list",
+)
+class List(AAZCommand):
+    """List all Extensions under ArcSetting resource.
+
+    :example: List Extensions under ArcSetting resource
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_list.py:20]:
```suggestion
    """List all extensions under arc setting resource.

    :example: List extensions under arc setting resource
```

File: src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_show.py [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "stack-hci extension show",
+)
+class Show(AAZCommand):
+    """Get particular Arc Extension of HCI Cluster.
+
+    :example: Get ArcSettings Extension
```
Comment (zhoxing-ms) [inline src/stack-hci/azext_stack_hci/aaz/latest/stack_hci/extension/_show.py:20]:
```suggestion
    """Get particular arc extension of HCI Cluster.

    :example: Get arc settings Extension
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**